ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh

```diff
@@ -395,27 +395,46 @@ oc wait clusteroperators --all --for=condition=Progressing=false --timeout=10m
 echo "$(date) - all clusteroperators are done progressing."
 
 # this works around a problem where tests fail because imagestreams aren't imported.  We see this happen for exec session.
-echo "$(date) - waiting for non-samples imagesteams to import..."
-count=0
+count=1
 while :
 do
-  non_imported_imagestreams=$(oc -n openshift get imagestreams -o go-template='{{range .items}}{{$namespace := .metadata.namespace}}{{$name := .metadata.name}}{{range .status.tags}}{{if not .items}}{{$namespace}}/{{$name}}:{{.tag}}{{"\n"}}{{end}}{{end}}{{end}}')
-  if [ -z "${non_imported_imagestreams}" ]
-  then
-    break
-  fi
-  echo "The following image streams are yet to be imported (attempt #${count}):"
-  echo "${non_imported_imagestreams}"
+  echo "[$(date)] waiting for non-samples imagesteams to import..."
+  wait_count=1
+  while :
+  do
+    non_imported_imagestreams=$(oc -n openshift get imagestreams -o go-template='{{range .items}}{{$namespace := .metadata.namespace}}{{$name := .metadata.name}}{{range .status.tags}}{{if not .items}}{{$namespace}}/{{$name}}:{{.tag}}{{"\n"}}{{end}}{{end}}{{end}}')
+    if [ -z "${non_imported_imagestreams}" ]
+    then
+      break 2 # break from outer loop
+    fi
+    echo "[$(date)] The following image streams are yet to be imported (attempt #${wait_count}):"
+    echo "${non_imported_imagestreams}"
 
+    wait_count=$((wait_count+1))
+    if (( wait_count > 10 )); then
+        break
+    fi
+
+    sleep 60
+  done
+
+  # Given up after 3 rounds of waiting 10 minutes
   count=$((count+1))
-  if (( count > 30 )); then
-    echo "Failed while waiting on imagestream import"
-    exit 1
+  if (( count > 3 )); then
+      echo "[$(date)] Failed to import all image streams after 30 minutes"
+      echo $non_imported_imagestreams
+      exit 1
   fi
 
-  sleep 60
+  # image streams won't retry by themselves https://issues.redhat.com/browse/RFE-3660
+  for imagestream in $non_imported_imagestreams
+  do
+      echo "[$(date)] Retrying image import $imagestream"
+      oc import-image "$imagestream"
```

Comment:
if this fails, it looks like the whole script will exit due to the `set -o errexit`, do you want that or do you want to `set +e` and `set -e` around this line so that you can see if the import-image error happens on specific images?